### DISCUSSION TITLE
Impute transform with frame parameter

### DISCUSSION THREAD

**Discussion by qnzhou** [vega/vega-lite#8766] [Q&A]
This is an followup from the discussion of this Atlair discussion thread: https://github.com/altair-viz/altair/discussions/2947

In summary, there seems to be a mismatch between the [documentation](https://vega.github.io/vega-lite/docs/impute.html) and the results we are getting from Altair/vega-lite. Here is a [minimum example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAbJQCMlAFYQ2AOxAAaEFBUAzGmTShiNOAHcDmlZhrKGbBhAAScPVjQBmAAweNW5VZt2EADqNAAm9GgALF4AvjEaoUiYKKigykgIcGggicn8AOxMcABMHnD5kaVQUGKFUG75AJyR+WXaMmLaHsX5AByNIPE5SUgQcJgQ5rkChSVlFVU1dQ3NrXDtnd19A6gA2qCYaN4gxEiCDFmoxRqnAE40SEyClyAAHoNqB2jXJ2cX7jckPdHs9su94l9UG5AedLjJAcCni9wZ8QIdUJEYf8oQiHkiwR9IcdTrC0GJcSCXgBPQlotDw36k1D5Cn49A0iF0q5Yy7k36I0Hs2nozGM7Giu54wUgDkAXSGcGUWlC1n0qQsgjYt3MujgglC2UllPUaKpOBeyjYCGsZw+bx1xn12UOGkwZpeAEcGEg-HQkjRSHaaerdU70CT-q73dkvT6rMkrIG4hpkLcANbmHBsazozC3C5R83ZYTKLJDPM+iDaLUIND7EDaW4ZS67eEyWUaMi3Ow4JjB3b8qVZDsgGgIHAMTAvCNZDRpuDBukp8YwNgG9CZH2DeVAA) to reproduce the problem (credit: @ChristopherDavisUCI).  

The x variable is missing the value corresponding to `t=3` and the y variable is missing the value corresponding to `t=1`.  With `"frame": [-1, 1]` and `"method": "mean"`, we are expecting `x=4.5` at `t=3` and `y=2.5` at `t=1`.  However, this is not the case as shown in the result plot (which indicates `x=4` and `y=3`). Any insight as to why this is the case will be very helpful!

**Comment by domoritz** [comment]
Is this an issue with Vega-Lite
or Vega?

**Reply by qnzhou** [reply]
Could be, but Vega's [impute documentation](https://vega.github.io/vega/docs/transforms/impute/) never mentions the support of a `frame` parameter.  On the other hand, `frame` is on the [official Vega-Lite doc](https://vega.github.io/vega-lite/docs/impute.html)...
